File: majority_element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0, ans = 0;

        for(auto num: nums) {
            if(count == 0) ans = num;

            if(num == ans) count++;
            else count--;
        }
        return ans;
    }
};

// Time Complexity : O(n)
// Space Complexity : O(1)

File: majority_element.go
func majorityElement(nums []int) int {
    ans, count := 0, 0

    for _, num := range nums {
        if count == 0 {
            ans = num
        }

        if num == ans {
            count++
        } else {
            count--
        }
    }
    return ans
}

// Time Complexity : O(n)
// Space COmplexity : O(1)